clap_derive/src/item.rs

```diff
@@ -903,7 +903,8 @@ impl Item {
             let short_name = format_ident!("{short_name}");
             let short = Method::new(
                 short_name,
-                short_help
+                long_help.clone()
+                    .or_else(|| short_help)
                     .map(|h| quote!(#h))
                     .unwrap_or_else(|| quote!(None)),
```

Comment:
This would cause the long help to be used for Arguments and Subcommands, and not just Value Enum Variants.